### PR TITLE
576 - Create and use loading label component and refactor

### PR DESCRIPTION
**Pull request by jamshale**
This is addressing issue 576 in the backlog. Thought I would pick up a smaller task for some practice.

Added a component that shows a spinner when a label is waiting to become defined. I'm not sure this would be visible very often because it would need to have the other stores loaded but not contacts. But anyway I think it looks decent and the component might be useful elsewhere.

I forced the contacts to delay loading for 3 seconds to see the effect.

[Traction Tenant Console.webm](https://github.com/bcgov/traction/assets/31809382/2703a910-b3bb-49cd-8daa-d4fe7f3decd4)

I refactored some code because the findByConnectionName function in the contacts store was only needed. Made it so if the contacts are loading findByConnectionName returns undefined type and if it can't find a match it returns a blank string (Maybe should should say not found or something other than blank ? Could be null but then would render blank anyway) 

Wrote some tests and some mocks. Found a couple annoying things and added gotchas in the readme and some comments.

I like to use VS code organize imports and sort lines so some imports go mix around.

### CODE REVIEW

**Review by loneil — CHANGES_REQUESTED** [review]
Other than comment about prop default/format it looks good to me, solution is pretty much what I had in mind for this. Tested locally (faked delay to see spinners).
Likely won't ever see these spinners much in real operations :) but it's a good thing to have if the connections ever load slowly (or slower than the main tables' datas)

Thanks again!

File: services/tenant-ui/frontend/src/components/common/LoadingLabel.vue [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+const props = defineProps({
+  value: {
+    type: String as PropType<string | undefined>,
+  },
+  size: {
+    type: Number as PropType<number>,
+    default: 1,
+  },
+});
```
[the remark this answers is not in the document]
Comment (jamshale) [reply services/tenant-ui/frontend/src/components/common/LoadingLabel.vue]: Done

**Review by loneil — APPROVED** [review]
Looks all good and I've tested out locally.

**Comment by jamshale** [comment]
> Looks all good and I've tested out locally. @jamshale can you just update the branch and then we will merge it.

Should be updated now.

**Comment by loneil** [comment]
> > Looks all good and I've tested out locally. @jamshale can you just update the branch and then we will merge it.
> 
> Should be updated now.

Thank you, merging to main.

**Pull request merged**